pkg/skaffold/build/kaniko/sources/localdir.go

```diff
@@ -0,0 +1,129 @@
+/*
+Copyright 2018 The Skaffold Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package sources
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"time"
+
+	"github.com/sirupsen/logrus"
+
+	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/constants"
+	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/docker"
+	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/schema/latest"
+	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/util"
+	"github.com/pkg/errors"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+)
+
+const (
+	initContainer = "kaniko-init-container"
+)
+
+// LocalDir refers to kaniko using a local directory as a buildcontext
+// skaffold copies the buildcontext into the local directory via kubectl cp
+type LocalDir struct {
+	tarPath string
+}
+
+// Setup for LocalDir creates a tarball of the buildcontext and stores it in /tmp
+func (g *LocalDir) Setup(ctx context.Context, artifact *latest.Artifact, cfg *latest.KanikoBuild, initialTag string) (string, error) {
+	g.tarPath = filepath.Join("/tmp", fmt.Sprintf("context-%s.tar.gz", initialTag))
+	logrus.Infof("storing buildcontext tarball at %s", g.tarPath)
+	f, err := os.Create(g.tarPath)
+	if err != nil {
+		return "", errors.Wrap(err, "creating temporary buildcontext tarball")
+	}
+	defer f.Close()
+	err = docker.CreateDockerTarGzContext(f, artifact.Workspace, artifact.DockerArtifact)
+	context := fmt.Sprintf("dir://%s", constants.DefaultKanikoEmptyDirMountPath)
+	return context, err
+}
+
+// Pod returns the pod template to ModifyPod
+func (g *LocalDir) Pod(cfg *latest.KanikoBuild, args []string) *v1.Pod {
+	p := podTemplate(cfg, args)
+	// Include the emptyDir volume and volume source in both containers
+	v := v1.Volume{
+		Name: constants.DefaultKanikoEmptyDirName,
+		VolumeSource: v1.VolumeSource{
+			EmptyDir: &v1.EmptyDirVolumeSource{},
+		},
+	}
+	vm := v1.VolumeMount{
+		Name:      constants.DefaultKanikoEmptyDirName,
+		MountPath: constants.DefaultKanikoEmptyDirMountPath,
+	}
+	// Generate the init container, which will run until the /tmp/complete file is created
+	ic := v1.Container{
+		Name:  initContainer,
+		Image: constants.DefaultDebianImage,
```

Comment:
Can we use something smaller like alpine? Debian seems a bit heavy. The only requirement is the tar binary right?

Reply:
fixed